src/Microsoft.DotNet.Build.Tasks/PackageFiles/packageresolve.targets

```diff
@@ -85,14 +87,18 @@
     <!-- We may have an indirect package reference that we want to replace with a project reference -->
     <ItemGroup>
       <!-- Intersect project-refs with package-refs -->
-      <_ReferenceFileNamesToRemove Include="@(Reference)" Condition="'@(_ResolvedProjectReferencePaths->'%(FileName)%(Extension)')' == '%(FileName)%(Extension)'" />
+      <_ReferenceFileNamesToRemove Include="@(Reference)" Condition="'$(_BuildAgainstPackages)' == 'true' and '@(ProjectReference->'%(FileName) - %(KeepProjectReference)')' == '%(FileName) - true'" />
+      <_ReferenceFileNamesToRemove Include="@(Reference)" Condition="'$(_BuildAgainstPackages)' != 'true' and '@(_ResolvedProjectReferencePaths->'%(FileName)%(Extension)')' == '%(FileName)%(Extension)'" />
```

Comment:
I'm trying to understand this change but I'm not entirely sure I follow the logic.